persistence-api/src/main/java/io/stargate/db/cdc/CDCServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cdc;
+
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import com.codahale.metrics.MetricRegistry;
+import com.datastax.oss.driver.shaded.guava.common.annotations.VisibleForTesting;
+import com.datastax.oss.driver.shaded.guava.common.util.concurrent.ThreadFactoryBuilder;
+import io.stargate.core.metrics.api.Metrics;
+import io.stargate.db.metrics.CDCMetrics;
+import java.util.concurrent.*;
+import org.apache.cassandra.stargate.db.MutationEvent;
+import org.apache.cassandra.stargate.exceptions.CDCWriteException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class CDCServiceImpl implements CDCService {
```

Comment:
Could it be preferable to move the "impl" service to different (new?) module so that the "API" module defines just java classes, but does not start services automatically?

Reply:
I think the service should be started automatically as it's possible to enable/disable cdc per table.
Ideally, a user will set up a CDC producer, deploy and later in time enable a table.

Reply:
About moving the service to a new module, I think we can discuss it after the query objects + persistence wrapper (C2-587) is merged. I think CDC service implementation should live wherever the persistence wrapper lives.